pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -880,40 +874,38 @@ func shouldImmediatelyFailMigration(vmi *v1.VirtualMachineInstance) bool {
 	return shouldFail
 }
 
-func (l *LibvirtDomainManager) asyncMigrate(vmi *v1.VirtualMachineInstance, options *cmdclient.MigrationOptions) error {
-
+func (l *LibvirtDomainManager) migrate(vmi *v1.VirtualMachineInstance, options *cmdclient.MigrationOptions) {
 	// get connection proxies for tunnelling migration through virt-handler
-	go func() {
-		if shouldImmediatelyFailMigration(vmi) {
-			log.Log.Object(vmi).Error("Live migration failed. Failure is forced by functional tests suite.")
-			l.setMigrationResult(vmi, true, "Failed migration to satisfy functional test condition", "")
-			return
-		}
 
-		migrationErrorChan := make(chan error, 1)
-		defer close(migrationErrorChan)
+	if shouldImmediatelyFailMigration(vmi) {
```

Comment:
Any reason we should do this check inside the go thread?  Why not on the caller function and avoid the calling of migrate completely ?